test/Conversion/HandshakeToFIRRTL/simple_addi.mlir

```diff
@@ -2,18 +2,31 @@
 
 // CHECK-LABEL: firrtl.module @handshake_merge_1ins_1outs(
 // CHECK-SAME:  %arg0: !firrtl.bundle<valid: uint<1>, ready: flip<uint<1>>, data: uint<64>>, %arg1: !firrtl.bundle<valid: flip<uint<1>>, ready: uint<1>, data: flip<uint<64>>>) {
-// CHECK:   %0 = firrtl.subfield %arg0("valid") : (!firrtl.bundle<valid: uint<1>, ready: flip<uint<1>>, data: uint<64>>) -> !firrtl.uint<1>
-// CHECK:   %1 = firrtl.subfield %arg0("ready") : (!firrtl.bundle<valid: uint<1>, ready: flip<uint<1>>, data: uint<64>>) -> !firrtl.flip<uint<1>>
-// CHECK:   %2 = firrtl.subfield %arg0("data") : (!firrtl.bundle<valid: uint<1>, ready: flip<uint<1>>, data: uint<64>>) -> !firrtl.uint<64>
-// CHECK:   %3 = firrtl.subfield %arg1("valid") : (!firrtl.bundle<valid: flip<uint<1>>, ready: uint<1>, data: flip<uint<64>>>) -> !firrtl.flip<uint<1>>
-// CHECK:   %4 = firrtl.subfield %arg1("ready") : (!firrtl.bundle<valid: flip<uint<1>>, ready: uint<1>, data: flip<uint<64>>>) -> !firrtl.uint<1>
-// CHECK:   %5 = firrtl.subfield %arg1("data") : (!firrtl.bundle<valid: flip<uint<1>>, ready: uint<1>, data: flip<uint<64>>>) -> !firrtl.flip<uint<64>>
-// CHECK:   firrtl.when %0 {
-// CHECK:     firrtl.connect %5, %2 : !firrtl.flip<uint<64>>, !firrtl.uint<64>
-// CHECK:     firrtl.connect %3, %0 : !firrtl.flip<uint<1>>, !firrtl.uint<1>
-// CHECK:     firrtl.connect %1, %4 : !firrtl.flip<uint<1>>, !firrtl.uint<1>
-// CHECK:   }
-// CHECK: }
+// CHECK:   %[[ARG0_VALID:.+]] = firrtl.subfield %arg0("valid") : (!firrtl.bundle<valid: uint<1>, ready: flip<uint<1>>, data: uint<64>>) -> !firrtl.uint<1>
```

Comment:
I see this test has some MergeOps, but they are just merging a single input. It seems to me this test case was generated from `circt-opt -create-dataflow` for a program that just does `std.addi %0, %1`. Such single input merges would just be canonicalized away. Should I remove them from this test case? Seems like lowering the merge isn't really what we are testing here, since there is a `test_merge.mlir` for that.

Reply:
Right, single input MergeOp will be eliminated after canonicalization. I think MergeOps in this test case can be removed.